src/db/z_db.erl

```diff
@@ -301,6 +311,7 @@ q1(Sql, Parameters, #context{} = Context) ->
 q1(Sql, #context{} = Context, Timeout) when is_integer(Timeout) ->
     q1(Sql, [], Context, Timeout).
 
+-spec q1(sql(), parameters(), #context{}, pos_integer()) -> term() | undefined.
```

Comment:
The spec in line 314 uses a record, please define a type for the record and use that instead.